### PR TITLE
[*] preview and publish bug fixed - added null check for content deletion

### PR DESCRIPTION
**Pull request by JackWCDS**

### CODE REVIEW

**Comment by JackWCDS** [comment]
adding test amendments

File: futurenhs.content.api/Umbraco9ContentApi.Test/Services/FutureNhsContentServiceTests.cs [hunk 1]
```diff
@@ -64,7 +67,7 @@ public void Setup()
         public void GetFutureNhsContentService()
         {
             _futureNhsContentService = new(_mockPublishedContentQuery.Object,
-_mockFutureNhsContentResolver.Object, _mockContentService.Object);
+_mockFutureNhsContentResolver.Object, _mockContentService.Object, _mockLogger.Object);
```
Comment (cds-tom-searle) [inline futurenhs.content.api/Umbraco9ContentApi.Test/Services/FutureNhsContentServiceTests.cs:70]: The indentation looks a little funky here, is this correct?

Comment (JackWCDS) [reply futurenhs.content.api/Umbraco9ContentApi.Test/Services/FutureNhsContentServiceTests.cs]: It does, corrected indentation

File: futurenhs.content.api/Umbraco9ContentApi.Core/Services/FutureNhs/FutureNhsContentService.cs [hunk 2]
```diff
@@ -79,9 +82,13 @@ public IContent CreateContent(string name, Guid parentId, string documentTypeAli
...
         {
             var content = _contentService.GetById(contentId);
-
-            if (!_contentService.Delete(content).Success)
-                throw new KeyNotFoundException($"Unable to delete content {contentId}. Content does not exist.");
+            if (content is not null)
+            {
+                if (!_contentService.Delete(content).Success)
+                {
+                    _logger.LogWarning($"Unable to delete content {contentId}. Content does not exist.");
```
[the remark this answers is not in the document]
Comment (JackWCDS) [reply futurenhs.content.api/Umbraco9ContentApi.Core/Services/FutureNhs/FutureNhsContentService.cs:89]: @cds-tom-searle what's your view?

[the remark this answers is not in the document]
Comment (JackWCDS) [reply futurenhs.content.api/Umbraco9ContentApi.Core/Services/FutureNhs/FutureNhsContentService.cs]: yes mate thank you

File: futurenhs.content.api/Umbraco9ContentApi.Core/Services/FutureNhs/FutureNhsContentService.cs [hunk 3]
```diff
@@ -79,9 +82,13 @@ public IContent CreateContent(string name, Guid parentId, string documentTypeAli
...
         {
             var content = _contentService.GetById(contentId);
-
-            if (!_contentService.Delete(content).Success)
-                throw new KeyNotFoundException($"Unable to delete content {contentId}. Content does not exist.");
+            if (content is not null)
+            {
+                if (!_contentService.Delete(content).Success)
+                {
+                    _logger.LogWarning("Unable to delete content {0}. Content does not exist.", contentId);
```
Comment (Matt-Beldon-CDS) [inline futurenhs.content.api/Umbraco9ContentApi.Core/Services/FutureNhs/FutureNhsContentService.cs:89]: sorry to be super pedantic but could the property name better describe parameter

**Pull request merged**